pkg/vsphere/session/session.go

```diff
@@ -155,14 +155,10 @@ func (s *Session) Connect(ctx context.Context) (*Session, error) {
 		}
 	}
 
+	soapURL.User = s.User
 	// Update the service URL with expanded defaults
 	s.Service = soapURL.String()
```

Comment:
It's really the same behavior as before; the only change is that we skip out on checking to make sure soapURL.User is non-nil because if it's nil we still want it added to the soapURL. @hickeng can weigh in on whether this might cause an issue elsewhere, hopefully.